### PR TITLE
Typed-column writer direct-view alignment and zero-validated reachability prefix-padding contracts

### PR DESCRIPTION
**Pull request by snissn**
## Objective

Implement writer-side typed-column-part fixed-width direct-view certification and deterministic padding contracts so certified fixed-width payloads are alignment-safe and accurately accounted in reachability.  
This now includes bounded deterministic prefix-padding reachability treatment for reserved direct-view segment refs and regular rewrite segment refs, with strict zero-byte validation before treating the bounded gap as known.

## Context

This change completes #1895 writer-side contract coverage under the #1886 workstream and incorporates follow-up review feedback on prefix-padding accounting safety. It preserves fail-closed behavior by allowing bounded prefix-padding inference only when the inferred gap is contiguous with prior coverage and verified as zero on disk.

## Non-goals

- Reader-side direct-view consumption APIs.
- Unsafe/direct memory-view query APIs.
- Dictionary/string direct-view layouts.
- Adjacency direct-view support (#1901).
- Physical-row-asset layout changes (#1897).
- Any change to fallback/deferred certification policy for non-eligible layouts.

## Design

- Formats/flags/APIs changed (include diagrams if applicable)
  - Writer metadata/certification behavior for typed-column-part fixed-width direct-view candidates.
  - Deterministic zero prefix padding for alignment (segment/appender paths) with explicit accounting.
  - Reachability bounded-window handling treats deterministic alignment prefix padding as known for:
    - reserved direct-view segment refs
    - regular rewrite segment refs (`tcs1_typed_column_part`)
  - Reachability inference no longer expands every aligned typed-column ref; it only marks the exact bounded prefix gap as known when that gap is directly after a previously covered range and validates as all-zero bytes on disk.
- Invariants that must hold
  - `DirectViewCertified` only for eligible declared fixed-width targets:
    - raw `int64`
    - native raw `float32`
    - native raw `float64`
    - fixed-dimension row-major `float32_vector`
  - Absolute offset alignment must hold for certified payloads.
  - Internal `__treedb_primary_id` must never be over-certified.
  - Deterministic prefix-padding known window remains bounded to alignment window (max 7 bytes).
- Fail-closed behavior (caps before alloc; CRC/error handling)
  - Non-eligible/ambiguous layouts remain uncertified and use fallback paths.
  - Non-zero bounded prefix gaps remain unknown.
  - Large unknown gaps remain unknown (not hidden by bounded padding accounting).

## Scope

- Includes (files/symbols):
  - Typed-column-part writer certification and alignment-padding logic.
  - Column-asset reachability bounds for deterministic prefix padding with bounded zero-byte validation.
  - Tests covering:
    - direct-view segment prefix-padding known window
    - rewrite segment prefix-padding known window
    - non-zero bounded prefix gap stays unknown
    - large-gap fail-closed unknown behavior
    - alignment and storage-accounting regressions
- Excludes (files/symbols):
  - Reader direct-view execution/query surfaces.
  - Column graph consumption changes.
  - New declared direct-view types beyond current fixed-width targets.
  - Adjacency/string/dictionary/nullable/default/compressed/delta certification expansion.

## Correctness

- Tests run (exact commands):
  - `GOWORK=off go test ./TreeDB/collections -run 'TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895|TestColumnAssetReachabilityRewriteSegmentPrefixPaddingIsKnown1895|TestColumnAssetReachabilityNonZeroPrefixPaddingGapStaysUnknown1895|TestColumnAssetReachabilityDirectViewPaddingWindowDoesNotHideLargeGaps1895|TestColumnAssetRewrite|TestColumnAssetGC' -count=1`
  - `GOWORK=off go test ./TreeDB/internal/typedcolumn ./TreeDB/internal/typeddecode ./TreeDB/internal/columnlayout ./TreeDB/collections -count=1`
  - `GOWORK=off go test ./TreeDB/...`
- Corruption/edge cases covered:
  - Bounded deterministic prefix-padding window is known only after zero-byte validation.
  - Non-zero bounded prefix padding stays unknown (fail-closed).
  - Larger reachability gaps stay unknown (fail-closed).
  - Internal primary-id non-certification regression remains covered.
- Concurrency/race coverage (if applicable):
  - No new concurrency model introduced; existing rewrite/GC/reachability paths validated by targeted tests above.

## Performance

- Benchmarks run (exact commands):
  - Not run (format/correctness-focused PR).
- Results table (before/after; median-of-5 per G1 in `docs/OPT_SPRINT_NEXT.md`):
  - Not applicable for this change set.
- Regressions (if any) and why acceptable:
  - None observed in targeted validation.

## Rollout / Toggle Plan

- Default setting:
  - Behavior is active in writer/reachability logic for eligible typed-column-part refs.
- How to disable quickly:
  - No new runtime toggle added; fallback behavior remains inherent for uncertified/non-eligible layouts.

## Follow-ups

- Issues/PRs to do next:
  - Reader-side direct-view consumption follow-ups from parent workstream.
  - Deferred layout support (dictionary/string/adjacency) in their respective follow-up issues.

### Copied/Adapted From Old Stack

- Copied:
  - N/A
- Adapted:
  - N/A
- Oracle/comparator only:
  - N/A
- Quarantined/not copied:
  - N/A

### Path Identity

- Native reader:
  - N/A (not in scope)
- Decoded comparator:
  - N/A
- Legacy/native vector index:
  - N/A
- Unsupported/fail-closed:
  - Unsupported layouts remain fail-closed for certification.

### Base And Dependency State

- Base branch:
  - `origin/main`
- Base commit:
  - `7dc8eb7c895ff198280660348ee9cd6ed0287241`
- Base PR/head:
  - Parent: #1886; branch head includes follow-up reachability fixes/tests.
- #1621/#1634 assumptions:
  - N/A
- Missing generic column-store primitives:
  - None newly introduced by this PR.
- Parent review fixes propagated:
  - Yes; includes primary-id over-certification guard and bounded zero-validated prefix-padding reachability updates for direct-view and rewrite refs.

### Evidence

- Tests:
  - See exact commands in **Correctness**.
- Benchmarks:
  - Not run (correctness/storage-format scope).
- Status/fallback proof:
  - Certification constrained to eligible fixed-width targets; bounded padding inference requires contiguous prior coverage plus zero-byte validation; others remain unknown/fallback.
- Performance attribution:
  - Not applicable.
- Local regressions found/fixed:
  - Reachability padding inference findings addressed with bounded known-window logic and non-zero-gap fail-closed regression coverage.

### Test Plan Start

- Milestone tasks targeted:
  - #1895 writer-side direct-view certification/alignment contracts.
- Tests to add before or with implementation:
  - Direct-view prefix-padding known-window + rewrite known-window + non-zero/large-gap fail-closed coverage.
- Existing tests to preserve:
  - Typed-column layout contract/certification and storage-accounting regressions.
- #1646 test-list changes made before implementation:
  - N/A

### Performance Plan Start

- Relevant benchmarks/metrics for this PR:
  - Not applicable (no fast-path enablement shipped).
- Baseline/comparator command or reason not applicable:
  - Not applicable.
- Expected setup/search/materialization boundary:
  - Reader/query fast paths out of scope.
- Upstream costs explicitly out of scope:
  - Query/vector direct-view performance.
- Local performance risks to watch:
  - None identified from current logic-only adjustments.

### Test Plan Close

- Additional tests found during implementation/review:
  - `TestColumnAssetReachabilityRewriteSegmentPrefixPaddingIsKnown1895`
  - `TestColumnAssetReachabilityNonZeroPrefixPaddingGapStaysUnknown1895`
- Tests added or changed:
  - Reachability regressions for deterministic prefix-padding known windows and preservation of fail-closed unknown behavior for non-zero/larger gaps.
- Failing tests fixed:
  - None newly introduced; targeted suites passed.
- #1646 test-list changes made at close:
  - N/A
- Residual untested gaps, if any:
  - None identified for #1895 scope.

### Performance Plan Close

- Benchmark commands run:
  - None.
- Results summary with throughput/latency/allocations:
  - Not applicable.
- Before/after or baseline comparison:
  - Not applicable.
- Local slow/heavy code found and fixed:
  - None.
- Remaining upstream or deferred costs:
  - Reader-side direct-view performance work remains deferred.

### AI Review Loop

- Codex latest-head review:
  - Addressed latest padding-inference finding by restricting bounded known-gap treatment to contiguous prior coverage with on-disk zero-byte validation; added non-zero-gap fail-closed regression.
- Copilot latest-head review:
  - Buffer-length clarity note fixed; thread resolved. No unresolved blocking comments.
- CodeRabbit latest-head review:
  - No blocking comments; rewrite reachability coverage assertions incorporated.
- Review comments/threads resolved or explicitly dismissed:
  - Reachability-prefix-padding threads resolved with bounded, zero-validated fail-closed behavior and targeted regressions.
- Re-requested after final meaningful push:
  - Yes.

---

### Scope
- [x] Single, clear objective for this PR (not a bundle)
- [x] Explicit include list (files/symbols touched):
- [x] Explicit exclude list (files/symbols NOT touched):
- [x] No unwanted feature reintroduction (e.g. async slab writer, hard zones, ValueIndex) unless explicitly stated in the PR objective

### Safety / Correctness
- [x] Clear written-vs-durable semantics for any `*Sync` path touched
- [x] No new mmap usage on mutable/truncating files
- [x] All new length fields are capped before allocation (OOM-safe)
- [x] CRC/checksum behavior is fail-closed (clean error, no panic)
- [x] Any concurrency change has a defined state machine and invariants

### Tests
- [x] Added deterministic regression tests for the targeted failure mode
- [x] Tests avoid sleeps where possible (use latches/hooks)
- [x] Added corruption/edge-case tests (truncation, bad headers, invalid lengths) when format/parsing changes
- [x] Ran locally:
  - [x] `GOWORK=off go test ./TreeDB/...`
  - [x] Any targeted packages/benchmarks:

### Benchmarks / Measurements
- [ ] Added or updated a benchmark relevant to the change
- [ ] Reported results in PR description (before/after, command, machine)
- [ ] Included “incompressible” (worst-case) results if compression is involved

### Docs
- [ ] Updated `docs/OPT_SPRINT_NEXT.md` milestone status (if applicable)
- [x] Documented any new config knobs + defaults
- [x] Called out any format change in PR description (pre-alpha OK)

### Rollout / Toggle Plan (if behavior-affecting)
- [x] Safe defaults (feature off or conservative threshold)
- [x] Clear knobs to disable/revert behavior quickly
- [x] Observability: counters/logs to verify effectiveness

## Summary by CodeRabbit

* **New Features**
  * Writers now insert deterministic zero-prefix padding so fixed-width typed-column payloads meet alignment rules.

* **Bug Fixes**
  * Stricter direct-view certification and alignment checks; misaligned or mask-containing columns fail closed.

* **Documentation**
  * Updated storage-format and typed-column spec docs describing certification, padding, and alignment rules.

* **Tests**
  * Extensive unit tests covering direct-view qualification, padding/alignment behavior, reachability accounting, and fallback cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/snissn/gomap/pull/1907?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by snissn** [comment]
@codex review

**Comment by snissn** [comment]
@copilot review

**Comment by snissn** [comment]
@coderabbitai review

**Comment by snissn** [comment]
Addressed Codex's reachability-padding finding in 017778d7f by extending deterministic typed-column direct-view prefix padding into reachability range coverage (bounded to the 7-byte alignment window) while preserving larger gaps as unknown. Added regressions for known alignment padding and large-gap fail-closed behavior.\n\nValidated locally:\n\n```sh\nGOWORK=off go test ./TreeDB/collections -run 'TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895|TestColumnAssetReachabilityDirectViewPaddingWindowDoesNotHideLargeGaps1895|TestColumnAssetTypedColumnPartDirectViewSegmentPaddingAlignment|TestColumnAssetTypedColumnPartDirectViewAppenderPaddingAlignment' -count=1\nGOWORK=off go test ./TreeDB/collections -run 'TestColumnAssetReachability|TestColumnAssetGC|TestColumnAssetRewrite' -count=1\nGOWORK=off go test ./TreeDB/internal/typedcolumn ./TreeDB/internal/typeddecode ./TreeDB/internal/columnlayout ./TreeDB/collections -count=1\n```

File: TreeDB/collections/column_asset_manager.go [hunk 1]
```diff
@@ -391,6 +392,18 @@ func writeColumnAssetToManagerSegment(rootDir string, cfg ColumnStoreConfig, pay
 	if err != nil {
 		return ColumnAssetRef{}, err
 	}
+	alignment := columnAssetSegmentPayloadAlignment(kind, cfg)
+	padding := columnAssetSegmentPrefixPadding(offset, alignment)
+	if padding > 0 {
+		written, err := writeColumnAssetSegmentZeroPadding(file, padding)
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (snissn) [reply TreeDB/collections/column_asset_manager.go:399]: Thanks. This is already handled in the current #1895 diff: reachability expands only direct-view typed-column-part refs by the deterministic prefix-padding window via `columnAssetReachabilityRangeBoundsForRef`/`columnAssetReachabilityRefHasDeterministicPrefixPadding`, so padding gaps are treated as known segment bytes rather than unknown corruption. The coverage is intentional and bounded to `ColumnAssetKindTCS1TypedColumnPart` refs in the direct-view segment band with aligned offsets. Regression coverage is in `TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895` (mixed protected/reclaimable segment with two padding gaps remains complete, `BytesUnknown == 0`) and `TestColumnAssetReachabilityDirectViewPaddingWindowDoesNotHideLargeGaps1895` (larger non-padding gaps still fail closed as unknown). No code change needed for this finding.

**Comment by snissn** [comment]
@codex review

**Comment by snissn** [comment]
@copilot review

**Comment by snissn** [comment]
@coderabbitai review

**Comment by snissn** [comment]
Final readiness update for latest head `4f9352f9bee60983fc565f435fe41efd2fa4a00f`:

- CI: all GitHub checks pass (`gh pr checks 1907`). Merge state is clean.
- Local focused/broad validation is listed in the PR body; full local suite had one unrelated `TreeDB/caching` scheduling failure that passed isolated rerun.
- Independent Orca review: PASS, no blocking findings after the primary-id certification fix.
- AI review status:
  - Copilot reviewed 12/12 files and generated no comments.
  - CodeRabbit generated no actionable comments; docstring coverage warning is non-blocking for this pre-alpha/internal storage PR.
  - Codex raised one reachability-padding accounting concern; answered with existing implementation/tests (`columnAssetReachabilityRangeBoundsForRef`, `TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895`, and large-gap fail-closed coverage). Thread is resolved.

Ready for human signoff/merge. I have not merged this PR.

**Comment by snissn** [comment]
Updated final readiness for latest head `017778d7ff6c92b60604d69d381b0ce939878742`:

- CI: all GitHub checks pass (`gh pr checks 1907`). Merge state is `CLEAN`.
- Local post-follow-up validation passed:
  - `GOWORK=off go test ./TreeDB/collections -run 'TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895|TestColumnAssetReachabilityDirectViewPaddingWindowDoesNotHideLargeGaps1895|TestColumnAssetTypedColumnPartFallbackSegmentsDoNotCertifyInternalPrimaryID|TestColumnAssetTypedColumnPartDirectViewSegmentPaddingAlignment|TestColumnAssetTypedColumnPartDirectViewAppenderPaddingAlignment|TestTypedColumnDirectViewWriterStorageAccounting1895|TestTypedStorageLegacyNameAllowlistIsComplete' -count=1`
  - `GOWORK=off go test ./TreeDB/internal/typedcolumn ./TreeDB/internal/typeddecode ./TreeDB/internal/columnlayout ./TreeDB/collections -count=1`
- AI reviews: Copilot no comments; CodeRabbit no actionable comments; Codex reachability-padding thread resolved against the latest reachability accounting/tests.

Ready for human signoff/merge. I have not merged this PR.

**Comment by snissn** [comment]
@codex review

**Comment by snissn** [comment]
@copilot review

**Comment by snissn** [comment]
Addressed the second Codex padding finding in b14d6abb8: reachability now treats the bounded deterministic prefix-padding window for aligned  refs as known in both reserved direct-view segment files and regular rewrite segment files. Added  to cover the regular rewrite segment case, while retaining the large-gap unknown fail-closed regression.\n\nValidated locally:\n\n```sh\nGOWORK=off go test ./TreeDB/collections -run 'TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895|TestColumnAssetReachabilityRewriteSegmentPrefixPaddingIsKnown1895|TestColumnAssetReachabilityDirectViewPaddingWindowDoesNotHideLargeGaps1895|TestColumnAssetRewrite|TestColumnAssetGC' -count=1\nGOWORK=off go test ./TreeDB/internal/typedcolumn ./TreeDB/internal/typeddecode ./TreeDB/internal/columnlayout ./TreeDB/collections -count=1\n```

**Comment by snissn** [comment]
@codex review

**Comment by snissn** [comment]
@copilot review

**Comment by snissn** [comment]
@coderabbitai review

**Comment by snissn** [comment]
Correction to the previous comment: the referenced ref kind is `tcs1_typed_column_part`, and the added rewrite regression is `TestColumnAssetReachabilityRewriteSegmentPrefixPaddingIsKnown1895`.

File: TreeDB/collections/column_asset_reachability_test.go [hunk 2]
```diff
@@ -1417,6 +1417,125 @@ func TestColumnAssetReachabilityPlanClassifiesFullCandidateSegmentReclaimableM15
...
+	if err != nil {
+		t.Fatalf("buildColumnAssetReachabilityPlan: %v", err)
+	}
+	if !plan.Complete || plan.Segments.Mixed != 1 || plan.Segments.Unknown != 0 || plan.Segments.BytesUnknown != 0 {
+		t.Fatalf("plan complete=%t segments=%+v want regular rewrite segment padding known", plan.Complete, plan.Segments)
+	}
+	if plan.Segments.BytesProtected != 33 || plan.Segments.BytesReclaimable != 16 || plan.RewriteDebtBytes != 16 {
+		t.Fatalf("segment bytes protected=%d reclaimable=%d rewrite=%d want protected=33 reclaimable=16 rewrite=16", plan.Segments.BytesProtected, plan.Segments.BytesReclaimable, plan.RewriteDebtBytes)
+	}
+}
```
[the remark this answers is not in the document]
Comment (snissn) [reply TreeDB/collections/column_asset_reachability_test.go:1508]: Fixed in `0115ae288` by adding the suggested segment-entry status/unknown-byte assertions and ref-level payload-only byte accounting checks to `TestColumnAssetReachabilityRewriteSegmentPrefixPaddingIsKnown1895`. Local focused and broad affected package tests passed on the new head.

**Comment by snissn** [comment]
@codex review

**Comment by snissn** [comment]
@copilot review

**Comment by snissn** [comment]
@coderabbitai review

**Comment by snissn** [comment]
Addressed CodeRabbit's rewrite reachability coverage suggestion in `0115ae288`: added assertions that the regular rewrite-segment padding test reports mixed segment details with zero unknown bytes and keeps ref accounting payload-only.

Validated locally:

```sh
GOWORK=off go test ./TreeDB/collections -run 'TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895|TestColumnAssetReachabilityRewriteSegmentPrefixPaddingIsKnown1895|TestColumnAssetReachabilityDirectViewPaddingWindowDoesNotHideLargeGaps1895|TestColumnAssetRewrite|TestColumnAssetGC' -count=1
GOWORK=off go test ./TreeDB/internal/typedcolumn ./TreeDB/internal/typeddecode ./TreeDB/internal/columnlayout ./TreeDB/collections -count=1
```

File: TreeDB/collections/column_asset_reachability.go [hunk 3]
```diff
@@ -1211,6 +1212,26 @@ func columnAssetReachabilityRefCanContributeRange(ref ColumnAssetRef, namespace
...
+			start = 0
+		}
+	}
+	return start, end
+}
+
+func columnAssetReachabilityRefHasDeterministicPrefixPadding(ref ColumnAssetRef) bool {
+	return ref.Kind == ColumnAssetKindTCS1TypedColumnPart &&
+		typedColumnPartDirectViewAssetAlignment > 1 &&
+		ref.Offset%int64(typedColumnPartDirectViewAssetAlignment) == 0
```
[the remark this answers is not in the document]
Comment (snissn) [reply TreeDB/collections/column_asset_reachability.go]: Fixed in `afe19891b`. Reachability no longer infers padding from alignment alone: it records original ref ranges, recognizes only exact inter-ref prefix-padding gaps for aligned `tcs1_typed_column_part` refs, and validates that the candidate gap bytes are zero before counting them as known coverage. Added `TestColumnAssetReachabilityNonZeroPrefixPaddingGapStaysUnknown1895` and tightened the large-gap test to keep the entire non-padding gap unknown/fail-closed. Local focused reachability/GC/rewrite and broad affected package tests passed.

**Comment by snissn** [comment]
@codex review

**Comment by snissn** [comment]
@copilot review

**Comment by snissn** [comment]
@coderabbitai review

**Comment by snissn** [comment]
Addressed the latest Codex padding-inference finding in `afe19891b`: reachability no longer expands every aligned typed-column ref. It now only treats the exact bounded prefix-padding gap as known when it follows a previous covered range and the on-disk gap bytes validate as zero; non-zero and larger gaps remain unknown/fail-closed.

Validated locally:

```sh
GOWORK=off go test ./TreeDB/collections -run 'TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895|TestColumnAssetReachabilityRewriteSegmentPrefixPaddingIsKnown1895|TestColumnAssetReachabilityNonZeroPrefixPaddingGapStaysUnknown1895|TestColumnAssetReachabilityDirectViewPaddingWindowDoesNotHideLargeGaps1895|TestColumnAssetRewrite|TestColumnAssetGC' -count=1
GOWORK=off go test ./TreeDB/internal/typedcolumn ./TreeDB/internal/typeddecode ./TreeDB/internal/columnlayout ./TreeDB/collections -count=1
```

File: TreeDB/collections/column_asset_reachability.go [hunk 4]
```diff
@@ -1211,6 +1225,77 @@ func columnAssetReachabilityRefCanContributeRange(ref ColumnAssetRef, namespace
...
+func columnAssetReachabilitySegmentRangeIsZero(path string, offset, length int64) bool {
+	if path == "" || offset < 0 || length <= 0 || length > int64(typedColumnPartDirectViewAssetAlignment-1) {
+		return false
+	}
+	file, err := os.Open(path)
+	if err != nil {
+		return false
+	}
+	defer func() { _ = file.Close() }()
+	buf := make([]byte, length)
```
[the remark this answers is not in the document]
Comment (snissn) [reply TreeDB/collections/column_asset_reachability.go]: Thanks. I verified this against Go: the `make` length argument accepts any integer type, not only `int`, and the latest head builds/tests successfully with this code. Local validation after `afe19891b` included the focused reachability tests plus `GOWORK=off go test ./TreeDB/internal/typedcolumn ./TreeDB/internal/typeddecode ./TreeDB/internal/columnlayout ./TreeDB/collections -count=1`; GitHub CI is also green. No code change needed.

**Comment by snissn** [comment]
Addressed Copilot's buffer-length clarity note in `96032983e` by converting the already bounded int64 padding length to `int` at the `make` call.

Validated locally:

```sh
GOWORK=off go test ./TreeDB/collections -run 'TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895|TestColumnAssetReachabilityRewriteSegmentPrefixPaddingIsKnown1895|TestColumnAssetReachabilityNonZeroPrefixPaddingGapStaysUnknown1895|TestColumnAssetReachabilityDirectViewPaddingWindowDoesNotHideLargeGaps1895|TestColumnAssetRewrite|TestColumnAssetGC' -count=1
```

**Comment by snissn** [comment]
@codex review

**Comment by snissn** [comment]
@copilot review

**Comment by snissn** [comment]
@coderabbitai review

**Comment by snissn** [comment]
@codex review

**Comment by snissn** [comment]
@copilot review

**Comment by snissn** [comment]
@coderabbitai review

**Comment by snissn** [comment]
Final readiness for latest head `96032983e2f75f0cfd5345d51fe09125ae8a4edd`:

- CI: all GitHub checks pass; merge state is `CLEAN` / mergeable.
- Local validation passed:
  - `GOWORK=off go test ./TreeDB/collections -run 'TestColumnAssetReachabilityDirectViewPrefixPaddingIsKnown1895|TestColumnAssetReachabilityRewriteSegmentPrefixPaddingIsKnown1895|TestColumnAssetReachabilityNonZeroPrefixPaddingGapStaysUnknown1895|TestColumnAssetReachabilityDirectViewPaddingWindowDoesNotHideLargeGaps1895|TestColumnAssetRewrite|TestColumnAssetGC' -count=1`
  - `GOWORK=off go test ./TreeDB/internal/typedcolumn ./TreeDB/internal/typeddecode ./TreeDB/internal/columnlayout ./TreeDB/collections -count=1`
  - `GOWORK=off go test ./TreeDB/...`
- AI review loop: Codex latest review found no major issues; Copilot buffer-length thread fixed/resolved; CodeRabbit has no unresolved actionable threads.
